#165598568 PR api bugs fix

PreciousTosin opened the pull request:
### What this PR does
- It fixes API bugs
#### Description of Task to be completed
- convert ES5 code to ES6
- ensure database tables have the proper reference
- add checks for proper error management
#### How it should be tested
 - follow the README.md guidelines to set up and run the application
 - use Postman to test the endpoints in the README
 - For authentication, use any of the test user accounts in the README documentation for login with this endpoint:
`POST localhost:9000/v1/auth/signin`
 - Use the bearer token option in the authentication tab. Supply the token generated.
#### Relevant pivotal tracker story
- [Delivers #165598568]

PreciousTosin committed:
Commit to make API fixes

bug(UI/css/style.css): fix API bugs
- install babel packages to transpile code
- Remove all ES5 Syntax
- use ES6 syntax in all files

- [Starts #165598568]

PreciousTosin committed:
Commit to fix coverage that fails

bug(package.json, .babelrc, .travis.yml): fix test coverage bug
- installed babel-instanbul, nyc, cross-env packages
- added nyc config to package.json
- update .babelrc file with test config

- [Delivers #165598568]

PreciousTosin committed:
Commit to make user route file scanty

bug(controller/user, model/user, test/user, controller/account): make user route file scanty
- move logic from user route to controller

- [Delivers #165598568]

PreciousTosin committed:
Commit to remove logic from transactions

bug(transaction models, controllers): fix API bugs
- work on transactions

- [Starts #165598568]

PreciousTosin committed:
Commit to fix failing travis build tests

bug(database/query.js): fix API test bugs
- work on making travis build succeed

- [Delivers #165598568]

augustineezinwa commented on database/controllers/account.js:
```diff
@@ -64,64 +57,95 @@ const accountController = {
+    const userAuthData = req.authData;
+    const { type } = req.body;
+    console.log('BANK ACCOUNT: ', type);
+    return new Promise((resolve, reject) => {
```
remove this `console.log`

PreciousTosin committed:
Commit to add validations for user inputs

bug(database/query.js): fix API bugs related to user input
- add validations for user signup and signin

- [Delivers #165598568]

PreciousTosin committed:
Commit to add validations for account, transaction endpoints

bug(models/*, controllers/*, routes/*, test/*): fix API bugs related to user input
- create validation functions for user, account, transaction
- add validations for user signup and signin, update
- add validations for account creation, update
- add validations for creating transactions
- remove all console.log messages
- add api prefix
- implement catchall route handler
- create new route for admin signup
- fix failing tests
- configure eslint to ignore build folder

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors
- remove build folder from gitignore
- remove postinstall script from package.json

- [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku build errors

bug(package.json): fix heroku build errors
- fix heroku errors
- remove build folder from gitignore
- remove postinstall script from package.json

- [Delivers #165598568]

PreciousTosin committed:
Commit to work on swagger docs

chore(package.json, server,js, route/*) add docs
 - Worked on swagger documentation

 [Delivers #165598568]

PreciousTosin committed:
Commit to update route authorization

feature(routes/*) update authorization
 - Update route authorization for all routes
 - recreated build

 [Delivers #165598568]

PreciousTosin committed:
Commit to fix heroku bug

feature(routes/*) fix heroku bug
 - rename npm build script

 [Delivers #165598568]
